File: searchingAlgorithms/linearSearch/linearsearch.cpp
#include <iostream>

using namespace std;

int arr[] = {4 , 5 , 3 , 17 , 8 , 6 , 13 , 45 , 7 , 88 , 56};

int arrayLen(int arr[]);
int linearsearch(int arr[] , int num , int size);

int main(){

	//cout << arrayLen(arr) << endl;

	int arrSize = sizeof(arr);

	int intSize = sizeof(arr[0]);

	int size = arrSize / intSize;

	int ansIndex = linearsearch(arr , 13 , size);

	if(ansIndex == -1){

		cout << "Number not found in array" << endl;

	}else{

		cout << "Number found at index " << ansIndex << endl;

	}

	return 0;

}

int linearsearch(int arr[] , int num , int size){

	int place = -1;

	cout << "size :" << size << endl;

	for (int  i = 0 ; i < size ; i++){

		if(arr[i] == num){

			place = i;

			break;

		}

	}

	return place;

}

File: searchingAlgorithms/linearSearch/linear_search.go
package main

import "fmt"
		

func main() {

	arr := []int{0,2,4,5,6,8}

	linearSearch(arr,6)
}

func linearSearch(arr []int, num int){
	var key int = num
	var found bool = false
	var index int = 0
	
	for i := 0; i < len(arr); i++ {
		if arr[i]==key {
			found = true
			index = i
		}
	}
	
	if found==true {
		fmt.Println("found!")
		fmt.Printf("Element found at Index:%d", index)
	}else{
		fmt.Printf("Element notfound")
	}
}


